packages/toast/src/Toast.tsx

```diff
@@ -0,0 +1,304 @@
+import React, { useRef } from 'react';
+import PropTypes from 'prop-types';
+import { Transition } from 'react-transition-group';
+import { transparentize } from 'polished';
+import { cx, css } from '@leafygreen-ui/emotion';
+import Portal from '@leafygreen-ui/portal';
+import { uiColors } from '@leafygreen-ui/palette';
+import { spacing } from '@leafygreen-ui/tokens';
+import { Body } from '@leafygreen-ui/typography';
+import IconButton from '@leafygreen-ui/icon-button';
+import CheckmarkWithCircleIcon from '@leafygreen-ui/icon/dist/CheckmarkWithCircle';
+import RefreshIcon from '@leafygreen-ui/icon/dist/Refresh';
+import XIcon from '@leafygreen-ui/icon/dist/X';
+import ProgressBar from './ToastProgressBar';
+
+const toastWidth = 400;
+
+const transitionDuration = 150;
+
+type StyledElements =
+  | 'toast'
+  | 'body'
+  | 'icon'
+  | 'contentWrapper'
+  | 'dismissButton';
+
+const baseElementStyles: Partial<Record<StyledElements, string>> = {
+  toast: css`
+    position: fixed;
+    bottom: ${spacing[6]}px;
+    left: ${spacing[4]}px;
+    width: ${toastWidth}px;
+    max-width: calc(100vw - ${spacing[4] * 2}px);
+    border-radius: 4px;
+    box-shadow: 0 18px 18px -15px ${transparentize(0.7, uiColors.black)};
+    overflow: hidden;
+    transform: translate3d(0, ${spacing[3]}px, 0) scale(0.95);
+    transform-origin: bottom center;
+    opacity: 0;
+    transition: all ${transitionDuration}ms ease-in-out;
+  `,
+
+  icon: css`
+    flex-shrink: 0;
+    margin-right: ${spacing[3]}px;
+  `,
+
+  contentWrapper: css`
+    display: flex;
+    align-items: center;
+    padding: ${spacing[3]}px;
+  `,
+
+  dismissButton: css`
+    position: absolute;
+    top: ${spacing[2]}px;
+    right: ${spacing[2]}px;
+    transition: color 0.15s ease-in-out;
+  `,
+};
+
+const ToastVariants = {
```

Comment:
`Button` and other components export their variants as `Variant`.

Reply:
Good point. I'll update this, but I'm not gonna export it for the time being.